lib/puppet/provider/iis_virtual_directory/webadministration.rb

```diff
@@ -18,47 +19,71 @@ def exists?
   def create
     Puppet.debug "Creating #{@resource[:name]}"
 
-    if @resource[:physicalpath].nil?
-      fail "physicalpath is a required paramter to create a iis virtual directory"
-    end
+    verify_physicalpath
 
-    if @resource[:physicalpath] and ! File.exists?(@resource[:physicalpath])
-      fail "physicalpath doesn't exist: #{@resource[:physicalpath]}"
-    end
+    # New-WebVirtualDirectory fails when PhysicalPath is a UNC path that is not available:
+    #   "Parameter 'PhysicalPath' should point to existing path."
 
-    cmd = []
-    cmd << "New-WebVirtualDirectory -Name \"#{@resource[:name]}\" "
-    cmd << "-Site \"#{@resource[:sitename]}\" " if @resource[:sitename]
-    cmd << "-Application \"#{@resource[:application]}\" " if @resource[:application]
-    cmd << "-PhysicalPath \"#{@resource[:physicalpath]}\" " if @resource[:physicalpath]
-    cmd << "-ErrorAction Stop"
-    cmd = cmd.join
+    if is_drive_path(@resource[:physicalpath])
+      cmd = []
+      cmd << "New-WebVirtualDirectory -Name \"#{@resource[:name]}\" "
```

Comment:
This would be more clear and easier to read if we decided which cmdlet to use instead of the entire block since most of the code is the same. For example, the unc block is missing ErrorAction stop.

Reply:
True